File: 067/main.cpp
#include <iostream>
#include <string.h>
using namespace std;

class Solution {
public:
    string addBinary(string a, string b) {
        int len1 = a.size();
        int len2 = b.size();
        int num;
        int dif;
        string res = "";
        string ss = "";
        if (len1 > len2) {
            num = len1;
            dif = len1 - len2;
            while (dif--) {
                b = "0" + b;
            }
        } else {
            num = len2;
            dif = len2 - len1;
            while (dif--) {
                a = "0" + a;
            }
        }
        cout << a << " " << b << endl;
        int flag = 0;
        int mid;
        for (int i = num - 1; i >= 0; i--) {
            mid = (a[i] - '0') + (b[i] - '0') + flag;
            if (mid >= 2) {
                mid -= 2;
                flag = 1;
            } else {
                flag = 0;
            }
            ss = mid + '0';
            res = ss + res;
        }
        if (flag) res = "1" + res;
        return res;
    }
};

int main()
{
    Solution sol;
    string a = "11";
    string b = "1";
    string res = sol.addBinary(a, b);
    cout << res << endl;
    return 0;
}

File: 067/main.js
/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function (a, b) {
  const len1 = a.length
  const len2 = b.length
  let num
  let str
  if (len1 > len2) {
    num = len2
    str = a
  } else {
    num = len1
    str = b
  }
  const s1 = a.split('').reverse().join('')
  const s2 = b.split('').reverse().join('')
  const su = str.split('').reverse().join('')
  let s3 = []
  let mid = 0
  let flag = 0
  for (let i = 0; i < num; i++) {
    mid = parseInt(s1[i]) + parseInt(s2[i]) + flag
    if (mid >= 2) {
      mid -= 2
      flag = 1
    } else {
      flag = 0
    }
    s3.push(mid)
  }
  for (let i = num; i < str.length; i++) {
    mid = parseInt(su[i]) + flag
    if (mid >= 2) {
      mid -= 2
      flag = 1
    } else {
      flag = 0
    }
    s3.push(mid)
  }
  if (flag) s3.push(1)
  return s3.reverse().join('')
}

console.log(addBinary('1010', '1011'))
console.log(addBinary('1111', '1111'))
